protocol/x/clob/keeper/deleveraging.go

```diff
@@ -499,23 +529,50 @@ func (k Keeper) ProcessDeleveraging(
 		),
 	)
 
-	// Send on-chain update for the deleveraging. The events are stored in a TransientStore which should be rolled-back
-	// if the branched state is discarded, so batching is not necessary.
-	k.GetIndexerEventManager().AddTxnEvent(
-		ctx,
-		indexerevents.SubtypeDeleveraging,
-		indexerevents.DeleveragingEventVersion,
-		indexer_manager.GetBytes(
-			indexerevents.NewDeleveragingEvent(
-				liquidatedSubaccountId,
-				offsettingSubaccountId,
-				perpetualId,
-				satypes.BaseQuantums(new(big.Int).Abs(deltaBaseQuantums).Uint64()),
-				satypes.BaseQuantums(deltaQuoteQuantums.Uint64()),
-				deltaBaseQuantums.Sign() > 0,
-			),
-		),
-	)
+	return nil
+}
+
+// DeleverageSubaccountsInFinalSettlementMarkets uses the subaccountOpenPositionInfo returned from the
+// liquidations daemon to deleverage subaccounts with open positions in final settlement markets. Note
+// this function will deleverage both negative TNC and non-negative TNC subaccounts. The deleveraging code
+// uses the bankruptcy price for the former and the oracle price for the latter.
+func (k Keeper) DeleverageSubaccountsInFinalSettlementMarkets(
```

Comment:
this is the function called in PrepareCheckState to deleverage positions in closed markets